recipes/full_finetune_distributed.py

```diff
@@ -273,18 +273,20 @@ def __init__(self, cfg: DictConfig) -> None:
         self.seed = training.set_seed(
             seed=cfg.seed, debug_mode=cfg.get("cudnn_deterministic_mode", None)
         )
-        self.epochs_run = 0
-        self.total_epochs = cfg.epochs
-        self.max_steps_per_epoch = cfg.max_steps_per_epoch
         self.global_step = 0
 
+        # Step-based training support
+        self.num_training_steps = cfg.num_training_steps
+        self._dataset_metrics_log_freq = cfg.get("dataset_metrics_log_freq", 100)
```

Comment:
Very minor but does this really need to be configured separately from log_every_n_steps? (Edit: I see you already added a TODO about this below, I guess you know my vote)